core/fileReference/ResourceFileIterator.php

```diff
@@ -0,0 +1,283 @@
+                if (!$remainingInstances) {
+                    $this->classIterator->next();
+                    $this->ensureNotEmpty();
+                }
+            }
+        }
+    }
+
+    /**
+     * While there are remaining classes there are instances to load
+     *
+     * @see Iterator::valid()
+     * @throws \common_exception_Error
+     */
+    public function valid()
+    {
+        if ($this->instanceCache === null) {
+            $this->ensureNotEmpty();
+        }
+
+        return $this->classIterator->valid();
+    }
+
+    /**
+     * Ensure the class iterator is pointin to a non empty class
+     * Loads the first resource block to test this
+     * @throws \common_exception_Error
+     */
+    protected function ensureNotEmpty()
+    {
+        $this->currentInstance = 0;
+        while ($this->classIterator->valid() && !$this->load()) {
+            $this->classIterator->next();
+        }
+    }
+
+    /**
+     * Load instances into cache
+     *
+     * @return boolean
+     * @throws \common_exception_Error
+     */
+    protected function load()
+    {
+        $results = $this->loadResources();
+        $this->instanceCache = [];
+        $this->currentInstance = 0;
+        foreach ($results as $resourceData) {
+            $this->instanceCache[] = $resourceData;
+        }
+
+        $this->endOfClass = count($results) < $this->cacheSize;
+
+        return count($results) > 0;
+    }
+
+    /**
+     * Load resources from storage
+     *
+     * @return mixed[][]
+     * @throws \common_exception_Error
+     */
+    protected function loadResources()
+    {
+        $fileResources = $this->getFileResources();
+        $parentData = $this->getFileParentResources($fileResources);
+
+        $resourcesData = [];
+        foreach ($fileResources as $resourceUri => $fileResource) {
+            $resourcesData[$resourceUri]['resource'] = new core_kernel_classes_Resource($resourceUri);
+            $resourcesData[$resourceUri]['id'] = $fileResource['id'];
+            if (!isset($parentData[$resourceUri])) {
+                $this->corruptFileIds[] = $fileResource['id'];
+                continue;
+            }
+            $resourcesData[$resourceUri]['property'] = $parentData[$resourceUri]['predicate'];
+            $resourcesData[$resourceUri]['parent'] = $parentData[$resourceUri]['subject'];
+        }
+
+        return $resourcesData;
+    }
+
+    /**
+     * Get the file resources
+     *
+     * @return mixed[]
+     */
+    private function getFileResources()
+    {
+        $sql = "
+            SELECT `subject`, `id` FROM ((
+                SELECT DISTINCT `id`, `subject` FROM `statements`
+                    WHERE  `predicate` = '" . OntologyRdf::RDF_TYPE . "'
+                    AND (`object` = '" . GenerisRdf::CLASS_GENERIS_FILE . "')
+                    AND `id` > " . $this->lastId . "
+                    AND `id` NOT IN('" . implode("', '", array_keys($this->corruptFileIds)) . "')
```

Comment:
Postgresql also doesn't like this: `SQLSTATE[22P02]: Invalid text representation: 7 ERROR:  invalid input syntax for integer: ""` It happens at the first run because `$this->corruptFileIds` is empty. Try to use something like this: https://github.com/oat-sa/tao-core/blob/master/models/classes/taskQueue/TaskLog/Broker/RdsTaskLogBroker.php#L353 but using `Connection::PARAM_INT_ARRAY` instead of `Connection::PARAM_STR_ARRAY`